src/platform_linux/platform_inline.h

```diff
@@ -150,22 +150,32 @@ platform_semaphore_try_wait(platform_semaphore *sema)
 }
 
 static inline platform_status
-platform_mutex_lock(platform_mutex *mu)
-{
-   int ret;
-
-   ret = pthread_mutex_lock(mu);
+platform_mutex_lock(platform_mutex *lock)
+{
+   int ret = pthread_mutex_lock(&lock->mutex);
+   platform_assert(
+      lock->owner == INVALID_TID,
+      "Locked a mutex with an existing owner, lock: %p, tid: %lu, owner: %lu\n",
+      lock,
+      platform_get_tid(),
+      lock->owner);
+   lock->owner = platform_get_tid();
+   platform_assert(lock->owner == platform_get_tid());
```

Comment:
Isn't this assert() redundant, especially considering the immediately preceding L162?

Are we suspecting some race here, which I don't see where it could occur, given that we have L155 where `pthread_mutex_lock() ` was acquired.

Seems like this line could be deleted? If you do wish to retain it, please do add a comment why it lives here.